weblate/api/serializers.py

```diff
@@ -923,6 +924,25 @@ class Meta:
         )
 
 
+class LabelsSerializer(serializers.RelatedField):
+    class Meta:
+        model = Label
+
+    def get_queryset(self):
+        unit = self.parent.parent.instance
+        project = unit.translation.component.project
+        return Label.objects.filter(project=project)
```

Comment:
Won't this always include all labels?

Also, not going through unit.labels makes it impossible to prefetch the query, causing an additional database query per each unit.

Something like this might work (completely untested):

```suggestion
        return self.parent.parent.instance.labels.all()
```

Reply:
It should only include the labels available in the project, right? Also I'm not sure how query caching works, so I have no idea on how to optimize for it.

Reply:
It should include only labels selected for the unit. Why the units endpoint should list all project labels?

Reply:
It doesn't, maybe you are misunderstanding what queryset does here: https://www.django-rest-framework.org/api-guide/relations/#the-queryset-argument

You don't want to limit the writable labels to only the ones that are already set.

Reply:
Sorry, I thought is used for reading. In that case, the code is correct. This might be easier to read:

```suggestion
        unit = self.parent.parent.instance
        project = unit.translation.component.project
        return project.label_set.all()
```

Reply:
I've merged my suggestion and added documentation to clarify when the queryset is used.

What is still missing are the tests.